smart_contracts.rst

```diff
@@ -154,7 +154,9 @@ Allows a channel participant to close the channel. The channel cannot be settled
 
 - ``channel_identifier``: Channel identifier assigned by the current contract.
 - ``nonce``: Strictly monotonic value used to order transfers.
+- ``balance_hash``: Hash of the balance data ``keccak256(transferred_amount, locked_amount, locksroot, additional_hash)``
```

Comment:
`additional_hash` is already blinding some data, this field is used to hide data exchanged among raiden nodes that is not needed on-chain but could be authenticated.

With this design, where the `additional_hash` is part of the `balance_hash`, a node that wants to authenticate the `data` hidden by `additional_hash` will also required the `balance_proof`, which could be good or not. 

An alternative design is to move the additional_hash "up" for the signature hash:

    hash(nonce || balance_hash || additional_hash || channel_identifier || token_network_address || chain_id)

I'm not sure how this plays with malleability though.